slam_toolbox/solvers/ceres_solver.cpp

```diff
@@ -322,28 +357,18 @@ void CeresSolver::AddConstraint(karto::Edge<karto::LocalizedRangeScan>* pEdge)
   // extract transformation
   karto::LinkInfo* pLinkInfo = (karto::LinkInfo*)(pEdge->GetLabel());
   karto::Pose2 diff = pLinkInfo->GetPoseDifference();
-  Eigen::Vector3d pose2d(diff.GetX(), diff.GetY(), diff.GetHeading());
-
-  karto::Matrix3 precisionMatrix = pLinkInfo->GetCovariance().Inverse();
-  Eigen::Matrix3d sqrt_information;
-  sqrt_information(0,0) = precisionMatrix(0,0);
-  sqrt_information(0,1) = sqrt_information(1,0) = precisionMatrix(0,1);
-  sqrt_information(0,2) = sqrt_information(2,0) = precisionMatrix(0,2);
-  sqrt_information(1,1) = precisionMatrix(1,1);
-  sqrt_information(1,2) = sqrt_information(2,1) = precisionMatrix(1,2);
-  sqrt_information(2,2) = precisionMatrix(2,2);
-
+  Eigen::Matrix3d sqrt_information =
+    pLinkInfo->GetCovariance().Inverse().ToEigen().llt().matrixL();
```

Comment:
Did we want to use the new one from #473 ?

Reply:
Yes! But I have to backport it to Noetic first.

Reply:
`Isbackported() {return true;};`

Reply:
Indeed, I've rebased this PR since.